### DISCUSSION TITLE
Can't figure out a package exclusion with multiple groups applied

### DISCUSSION THREAD

**Discussion by jraillard** [renovatebot/renovate#30779] [Request Help]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitLab

### Please tell us more about your question or problem

I would like to make some packageRules in order to have the following groups : 

- all dependencies major updates => all-deps-major  MR branch name
- all dependencies minor & patch updates => all-deps-minor-patch MR branch name
- internal framework major updates => all-internal-major MR branch name
- internal framework moinor & patch updates => all-internal-minor-patch MR branch name
- exclude some specific package 
=> for example : EPPlus should stay in 4.5.3.3 (free license)

With the following renovate.json I figured out almost all my needs but despite all my tentatives (ignoreDeps, packageRules with packagePackageNames & enabled false, ignorePaths, excludeNamePatterns, etc.) I wasn't able to exclude this damn EPPlus package from my renovate MR 
=> As a result it always appears in all-deps-major MR as there is a 7.2.2 version available

```json 
{
    "$schema": "https://docs.renovatebot.com/renovate-schema.json",
    "extends": [
        "config:best-practices",
        ":prConcurrentLimitNone",
        ":prHourlyLimitNone",
        ":prImmediately",
        ":rebaseStalePrs",
        ":semanticCommits",
        ":updateNotScheduled"
    ],    
    "automerge": false,
    "packageRules": [
        {
            "groupName": "All dependencies Major",
            "groupSlug": "all-deps-major",
            "assignees": [
                "jraillard"
            ],
            "matchPackageNames": [
                "*"
            ],
            "matchUpdateTypes": [
                "major"
            ]
        },
        {
            "groupName": "All dependencies Minor & Patch",
            "groupSlug": "all-deps-minor-patch",
            "assignees": [
                "jraillard"
            ],
            "matchPackageNames": [
                "*"
            ],
            "matchUpdateTypes": [
                "minor",
                "patch"
            ]
        },
        {
            "groupName": "Packages Internal Minor & Patch",
            "groupSlug": "all-internal-minor-patch",
            "assignees": [
                "jraillard"
            ],
            "matchPackageNames": [
                "company.internal.**"
            ],
            "matchUpdateTypes": [
                "minor",
                "patch"
            ]
        },
        {
            "groupName": "Packages Internal Major",
            "groupSlug": "all-internal-major",
            "assignees": [
                "jraillard"
            ],
            "matchPackageNames": [
                "company.internal.**"
            ],
            "matchUpdateTypes": [
                "major"
            ]
        }
    ]    
}
```

The solution should allow me to not specifically exclude this package in every groups as i would like to export it later as presets and that the "ignored package" list could grow.

Really hope this issue will find some lovely supports 🙏 

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by jraillard** [comment]
If it can help, i'm using renovate with version 38.37.1

**Reply by viceice** [reply]
add a new package rule for epplus and set enabled to false

**Reply by jraillard** [reply]
I finally found the problem. 
I didn't mentionned it but i put the property => `baseBranches: [" config-renovate "]`
That way i wanted to update the config in that branch and don't polluate main with my tests.
Unfortunately this config means : 
> get the renovate config from default branch, and apply the rules on `baseBranches`

==> One of the way (that i find a bit tricky) to edit the config is to nuke the "welcome PR config" mentionned here : https://docs.renovatebot.com/getting-started/installing-onboarding/#nuke-config-and-re-onboard

As a proposition, it would be nice to get a prop let's says  `baseConfigBranch` of type string that would be filled with git defaultBranch by default in order to get the renovate.json file
